src/components/Provider.js

```diff
@@ -20,7 +20,7 @@ function warnAboutReceivingStore() {
 
 export default class Provider extends Component {
   getChildContext() {
-    return { store: this.store }
+    return { store: this.store, react15CompatibilityMode: this.props.react15CompatibilityMode }
```

Comment:
Correct, and for 5.0 it will be opt-out of the slower (but more compatible) behavior. For 5.1, which will wait on React 16, it will be opt-in for those who need to stick with React <=15.